testsuite/tests/typing-layouts/allow_any.ml

```diff
@@ -0,0 +1,180 @@
+   shouldn't get an error (though, obviously, this is completely unsound!) *)
+module Mismatched_with_both_attrs : sig
+  type t : float64
+  [@@unsafe_allow_any_kind_in_impl "I love segfaults"]
+end = struct
+  type t = string
+  [@@unsafe_allow_any_kind_in_intf "I love segfaults"]
+end
+[%%expect{|
+module Mismatched_with_both_attrs : sig type t : float64 end
+|}]
+
+(* If we set the attributes but *don't* get a kind mismatch, we ought to be fine *)
+(* CR-soon aspsmith: this should be a warning *)
+module Matching : sig
+  type t : value
+  [@@unsafe_allow_any_kind_in_impl "I love segfaults"]
+end = struct
+  type t = string
+  [@@unsafe_allow_any_kind_in_intf "I love segfaults"]
+end
+[%%expect{|
+Lines 2-3, characters 2-54:
+2 | ..type t : value
+3 |   [@@unsafe_allow_any_kind_in_impl "I love segfaults"]
+Warning 212: [@@allow_any_kind_in_intf] and [@@allow_any_kind_in_impl] set on a
+type, but the kind matches. The attributes can be removed.
+
+module Matching : sig type t end
+|}]
+
+(* If the attr is only on the signature we should get an error *)
+module Mismatched_with_attr_on_intf : sig
+  type t : float64
+  [@@unsafe_allow_any_kind_in_impl "I love segfaults"]
+end = struct
+  type t = string
+end
+[%%expect{|
+Lines 4-6, characters 6-3:
+4 | ......struct
+5 |   type t = string
+6 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type t = string end
+       is not included in
+         sig type t : float64 end
+       Type declarations do not match:
+         type t = string
+       is not included in
+         type t : float64
+       The layout of the first is value
+         because it is the primitive type string.
+       But the layout of the first must be a sublayout of float64
+         because of the definition of t at lines 2-3, characters 2-54.
+|}]
+
+(* If the attr is only on the struct we should get an error *)
+module Mismatched_with_attr_on_impl : sig
+  type t : float64
+end = struct
+  type t = string
+  [@@unsafe_allow_any_kind_in_intf "I love segfaults"]
+end
+[%%expect{|
+Lines 3-6, characters 6-3:
+3 | ......struct
+4 |   type t = string
+5 |   [@@unsafe_allow_any_kind_in_intf "I love segfaults"]
+6 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type t = string end
+       is not included in
+         sig type t : float64 end
+       Type declarations do not match:
+         type t = string
+       is not included in
+         type t : float64
+       The layout of the first is value
+         because it is the primitive type string.
+       But the layout of the first must be a sublayout of float64
+         because of the definition of t at line 2, characters 2-18.
+|}]
+
+(* Some more complex stuff with functors *)
+
+module type S1 = sig
+  type t : value
+end
+
+module type S2 = sig
+  type t : float64
+  [@@unsafe_allow_any_kind_in_impl]
+end
+
+module F (M : S1) : S2 = struct
+  type t = M.t
+  [@@unsafe_allow_any_kind_in_intf]
+end
```

Comment:
I don’t think this test is as interesting as others: it still tests a `struct` against a `sig`. Instead, I was thinking of something like

```
module type S1 = sig
  type t : value
  [@@unsafe_allow_any_kind_in_intf]
end

module F (X : S1) : S2 = X (* should work *)
module F (X : S2) : S1 = X (* shouldn’t work *)
```